samples/application_development/number_crunching/README.rst

```diff
@@ -0,0 +1,130 @@
+.. _number_crunching:
+
+Number Crunching
+#####################
+
+Overview
+********
+
+Number crunching sample does vector operations,
+Fast Fourier Transformation and filtering.
+This example demonstrates how to include a proprietary
+static library into the Zephyr build system.
+The library is in an out of tree location.
+
+To use this sample, with an out of tree library, one needs to define
+an environnent variable `LIB_LOCATION`.
+In that location, one needs to have a `CMakeLists.txt` in which
+it defines:
+
+.. code-block:: console
+
+	# Link with the external 3rd party library.
+	set(LIB_DIR         "lib"                   CACHE STRING "")
+	set(INCLUDE_DIR     "include"               CACHE STRING "")
+	set(LIB_NAME        "proprietary_lib.a"     CACHE STRING "")
+
+If the environment variable ``LIB_LOCATION`` is not defined, a default Zephyr API
+is used instead.
+
+This sample executes some mathematical functions that can be used for audio
+processing like filtering (Fast Fourier Transform (FFT)) or echo cancellation
+(Least Mean Square (LMS) filter algorithm).
+
+The sample has:
+
+- :file:`main.c`: calls the generic math functions;
+- :file:`math_ops.c`: executes the math functions, computes the cycles it took
+  to execute and checks the output;
+- :file:`cmsis_dsp_wrapper.c`: calls the exact math functions from CMSIS-DSP
+  if ``CONFIG_CMSIS_DSP`` is defined and ``LIB_LOCATION`` is not defined;
+- :file:`nature_dsp_wrapper`: if ``LIB_LOCATION`` is defined and points to an out of tree
+  location where that NatureDSP lib and headers can be found, calls the exact math
+  functions from NatureDSP library.
```

Comment:
this is problematic as this comes with hardly any context. Yes, the point of the sample is probably to showcase proprietary/3rd party integration, but as such the sample will be of little use to folks (myself included) not familiar with what NatureDSP is in the first place. Please see my overrall "review comment" for suggested alternative.

Reply:
I've moved the sample in samples/modules/cmsis_dsp/number_crunching/.
NatureDSP is the CMSIS-DSP for Xtensa. This can be taken from here: https://github.com/foss-xtensa/ndsplib-hifi4/tree/main, compiled and linked with this sample as I explained in the readme.